rshim.c

```diff
@@ -2765,6 +2610,96 @@ static const struct cuse_methods rshim_misc_fops = {
 };
 #endif
 
+/* Rshim file operations routines */
+
+/* ioctl message header. */
+typedef struct {
+  uint32_t addr;
+  uint64_t data;
+} __attribute__((packed)) rshim_ioctl_msg;
+
+#ifdef HAVE_RSHIM_FUSE
+enum {
+  RSHIM_IOC_READ = _IO('R', 0),
```

Comment:
Size of IOCTL argument must match:
RSHIM_IOC_READ = _IOWR('R', 0, rshim_ioctl_msg)
RSHIM_IOC_WRITE = _IOWR('R', 1, rshim_ioctl_msg)

Reply:
Updated. Could you check the new content?